compile/service.go

```diff
@@ -313,7 +313,12 @@ func (rs *ResultSpec) Link(scope Scope) (err error) {
 	// verify that everything listed under throws is an exception.
 	for _, exception := range rs.Exceptions {
 		spec, ok := exception.Type.(*StructSpec)
-		if !ok || spec.Type != ast.ExceptionType {
+		if !ok {
+			return notAnExceptionError{
+				FieldName: exception.ThriftName(),
+			}
+		}
+		if spec.Type != ast.ExceptionType {
```

Comment:
I think you can collapse the original if statement back but use
`exception.Type.ThriftName()` instead of `spec.ThriftName()`.